validator/sawtooth_validator/journal/publisher.py

```diff
@@ -125,10 +126,6 @@ def _build_block(self, chain_head):
         self._scheduler = self._transaction_executor.create_scheduler(
             self._squash_handler, chain_head.state_root_hash)
 
-        self._transaction_executor.execute(self._scheduler)
-        for batch in self._pending_batches:
-            self._validate_batch(batch)
-
         # build the TransactionCache
         self._committed_txn = TransactionCache(self._block_cache.block_store)
```

Comment:
I don't see the double set here, The TransactionCache is being rebuilt to be in the state of the new block and with the addition of transactinons of a block not yet in the chain.